src/DIRAC/RequestManagementSystem/DB/RequestDB.py

```diff
@@ -869,6 +869,16 @@ def getRequestStatus(self, requestID):
             session.close()
         return S_OK(status[0])
 
+    def getBulkRequestStatus(self, requestIDs):
+        """get requests statuses for given request IDs"""
+        session = self.DBSession()
+        try:
+            statuses = session.query(Request.RequestID, Request._Status).filter(Request.RequestID.in_(requestIDs)).all()
+            status_dict = {req_id: req_status for req_id, req_status in statuses}
+        finally:
```

Comment:
The couple of functions above catch exceptions.

```suggestion
        except Exception as e:
            # log as well?
            return S_ERROR(f"Failed to getBulkRequestStatus {e!r}")
        finally:
```